components/datetime/src/format/neo.rs

```diff
@@ -2031,139 +2034,175 @@ impl<R: DateTimeNamesMarker> RawDateTimeNames<R> {
+                        mz_period_provider,
+                        locale,
+                    )?;
+                    // For fallback:
+                    self.load_time_zone_location_names(locations_provider, locale)?;
+                }
+
+                // V
+                (FS::TimeZone(TimeZone::Location), One) => {
+                    // no data required
+                }
+                // VVVV
+                (FS::TimeZone(TimeZone::Location), Wide) => {
+                    load_fdf = true;
+                    self.load_time_zone_essentials(zone_essentials_provider, locale)?;
+                    self.load_time_zone_location_names(locations_provider, locale)?;
+                }
+
+                // O, OOOO
+                (FS::TimeZone(TimeZone::LocalizedOffset), One | Wide) => {
+                    self.load_time_zone_essentials(zone_essentials_provider, locale)?;
+                    load_fdf = true;
+                }
+
+                // X..XXXXX, x..xxxxx
+                (
+                    FS::TimeZone(TimeZone::IsoWithZ | TimeZone::Iso),
+                    One | TwoDigit | Abbreviated | Wide | Narrow,
+                ) => {
+                    // no data required
+                }
+
                 ///// Numeric symbols /////
-                FieldSymbol::Year(_) => numeric_field = Some(field),
-                FieldSymbol::Week(_) => numeric_field = Some(field),
-                FieldSymbol::Day(_) => numeric_field = Some(field),
-                FieldSymbol::Hour(_) => numeric_field = Some(field),
-                FieldSymbol::Minute => numeric_field = Some(field),
-                FieldSymbol::Second(_) => numeric_field = Some(field),
-                FieldSymbol::DecimalSecond(_) => numeric_field = Some(field),
+
+                // y+, r+
+                (FS::Year(Year::Calendar | Year::RelatedIso), _) => load_fdf = true,
+
+                // M..MM, L..LL
+                (FS::Month(_), One | TwoDigit) => load_fdf = true,
+
+                // e..ee, c..cc
+                (FS::Weekday(Weekday::Local | Weekday::StandAlone), One | TwoDigit) => {
+                    load_fdf = true
+                }
+
+                // d..dd
+                (FS::Day(Day::DayOfMonth), One | TwoDigit) => load_fdf = true,
+                // D..DDD
+                (FS::Day(Day::DayOfYear), One | TwoDigit | Abbreviated) => load_fdf = true,
+                // F
+                (FS::Day(Day::DayOfWeekInMonth), One) => load_fdf = true,
+
+                // K..KK, h..hh, H..HH, k..kk
+                (FS::Hour(_), One | TwoDigit) => load_fdf = true,
+
+                // m..mm
+                (FS::Minute, One | TwoDigit) => load_fdf = true,
+
+                // s..ss
+                (FS::Second(Second::Second), One | TwoDigit) => load_fdf = true,
+
+                // A
+                (FS::Second(Second::Millisecond), _) => load_fdf = true,
+
+                // ???
```

Comment:
nit: ?????

(more docs please if this isn't following the regular scheme)

Reply:
yeah @sffc, I don't understand where this is coming from. There's no symbol associated with that type, and "decimal second" is not a UTS-35 concept

Reply:
I'm happy with this landing for now with someone doing the comment fix in a followup, but probably worth waiting for @sffc review too

Reply:
DecimalSecond was added in https://github.com/unicode-org/icu4x/pull/5392 and it is a single field to represent things like `ss.SSS` in UTS 35 patterns. Previously it was handled with hacky, buggy code.